### PR TITLE
Course Sidebar module fix

### PR DESCRIPTION
**Pull request by wangqiang1208**
Now if there were no modules included, the sidebar function will return nil, which is not expected.

### CODE REVIEW

File: spec/controllers/course/controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe Course::Controller, type: :controller do
+  describe '#sidebar' do
+    it 'returns an empty array when no modules included' do
+      expect(Course::Controller.new.sidebar).to eq([])
```
Comment (wangqiang1208) [inline spec/controllers/course/controller_spec.rb]: Actually my specs here is a bit nasty.. any other way to get the instance of  controller ?

File: spec/controllers/course/controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Course::Controller, type: :controller do
+  describe '#sidebar' do
+    it 'returns an empty array when no modules included' do
+      allow(Course::CoursesController).to receive(:modules).and_return([])
+      expect(Course::Controller.new.sidebar).to eq([])
```
Comment (wangqiang1208) [inline spec/controllers/course/controller_spec.rb]: Any other way that we can get the instance of controller to test?

Comment (lowjoel) [inline spec/controllers/course/controller_spec.rb]: `expect(controller.sidebar).to eq([])`

File: spec/controllers/course/controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Course::Controller, type: :controller do
+  describe '#sidebar' do
+    it 'returns an empty array when no modules included' do
+      allow(Course::CoursesController).to receive(:modules).and_return([])
+      expect(controller.sidebar).to eq([])
```
Comment (wangqiang1208) [inline spec/controllers/course/controller_spec.rb]: Don't even know Rspec provided us a controller here ><

**Pull request merged**